### DISCUSSION TITLE
Func/Lamba support?

### DISCUSSION THREAD

**Discussion by CoenraadS** [LuckyPennySoftware/MediatR#737] [Q&A]
Hi, I haven't used this package yet, but was just reading the wiki about it's implementation and this was not clear to me.

It it possible to do something like:

```
IMediator mediator;

using var listenerLifetime = mediator.Handle(async (SomeMessage) e => await { ... })
// On Dispose, unregisters listener
```

**Comment by jbogard** [comment]
No, you have to create a handler class. I'm not sure what the use case for the above would be. Just to get all the pipeline behaviors to run?

**Reply by CoenraadS** [reply]
I'm looking to see if it's feasible to replace our custom messaging solution, which is very similar to MediatR, except for this missing functionality.

Basically our project is a mix of OOP and Functional programming styles.

We have patterns like

```
{
  using var _ = mediator.Handle(async (SomeMessage) e => await { ... })
  foo.DoSomethingThatGeneratesMessages();
}
```

We sometimes don't want classes listening to messages all of the time, only in the short period that they are actually required to be listening. It makes code much easier to reason about, and listener code can live closely to message generating code.

**Reply by jbogard** [reply]
The classes can be defined anywhere, and in MediatR, nothing is "listening" to messages. Each time you send a request, the Mediator class delegates to an `IServiceProvider` to provide an instance of the handler. In typical scenarios these classes are instantiated and disposed with each Send/Publish.

In any case, MediatR is intentionally class/object-oriented, it's likely not a good fit in a functional-style application.
